classes/task/emit_task.php

```diff
@@ -59,11 +59,45 @@ private function get_event_ids($loadedevents) {
 
     private function extract_events($limitnum) {
         global $DB;
+
         $conditions = null;
         $sort = 'id DESC';
         $fields = '*';
         $limitfrom = 0;
-        $extractedevents = $DB->get_records('logstore_xapi_log', $conditions, $sort, $fields, $limitfrom, $limitnum);
+        $params = array();
+
+        $sql = "SELECT $fields
+                  FROM {logstore_xapi_log}";
+
+        $excludeusers = get_config('logstore_xapi', 'excludeusers');
+
+        if (!empty($excludeusers)) {
+
+            $excludeduserids = array_filter(explode(',', $excludeusers));
+
+            list($conditions, $params) = $DB->get_in_or_equal($excludeduserids, SQL_PARAMS_NAMED, 'userid');
+
+            // Construct Where cause.
+            if (count($excludeduserids) === 1) {
+                $sql .= " WHERE userid !". $conditions;
+            } else {
+                $sql .= " WHERE userid NOT ". $conditions;
+            }
+
+            // Delete only excluded users log.
+            $cleanupsql = '';
+            $cleanupsql = "DELETE FROM {logstore_xapi_log}";
```

Comment:
This suggests to me that we're still logging events for excluded users to the XAPI table. Why? Why not just disable them from logging entirely, rather than after-the-fact deleting them?

Reply:
Just from what I've discussed with Heena in DMs... after a quick look, I would guess that the `is_event_ignored()` function in classes/log/store.php` could be extended.

I'm assuming all logging goes through that function (though I haven't checked, so that would need to be done as well), and I also assume the `$event` has the userid (I wouldn't think of it much of an event, otherwise), so we should be able to check for excluded users here, and `return true` if need be.

Reply:
Example psuedo-code: `if ($event->userid IN $excludeusers) { return true; }`

Reply:
Thanks Joel, I've excluded from logging it entirely.

Reply:
This looks good now.